### DISCUSSION TITLE
Saving a vue file formats it

### DISCUSSION THREAD

**Discussion by pstaabp** [vuejs/language-tools#1014] [Q&A]
Just switched over to volar and am having a strange problem. When I save a vue file, it seems to format even though I explicitly have formatOnSave set to false.  This would be fine, if it was picking up the right formatting.  I have quotes set to single and indent to tab in `.eslintrc.js`  The save formats to spaces and string to double quotes.  Perhaps volar is picking up the settings from somewhere else?

This has be a setting because I have a different project that I started from scratch and it is not trying to format on save.  However if I do put format on save in this project, it uses double quotes and spaces for indent. 

Here's my questions:
1. Any reason why vscode is ignoring the formatOnSave option on a project is being ignored? I've only noticed this when using volar.
2. Where is volar looking for formatting?  I was assuming it would be eslint, but it doesn't seem to be.

**Comment by johnsoncodehk** [comment]
Volar doesn't perform extra formatting, the problem seems to be similar to #425, please check if https://github.com/johnsoncodehk/volar/issues/425#issuecomment-923189656 solves your problem.

**Reply by pstaabp** [reply]
I'm not using headwind. This may not be a volar issue.  For some reason vscode is running format on save, even if it's not requested.

**Reply by johnsoncodehk** [reply]
Can you reproduce the problem when disabled volar?

**Reply by pstaabp** [reply]
I disabled volar and the volar extension pack and I'm not getting the formatOnSave problems I mentioned. 

I reenabled volar (but not the extension pack) and it is not formatting on save.  This now gives me the correct behavior.   Seems to be a problem with the extension pack (or one of the extension within it).

**Comment by xiaoxiangmoe** [comment]
Can you provide a minimal reproduction? I'll try it.
